bot/exts/help_channels/_cog.py

```diff
@@ -535,3 +539,93 @@ async def update_available_help_channels(self) -> None:
         )
         self.dynamic_message = new_dynamic_message["id"]
         await _caches.dynamic_message.set("message_id", self.dynamic_message)
+
+    @staticmethod
+    def _serialise_session_participants(participants: set[int]) -> str:
+        """Convert a set to a comma separated string."""
+        return ','.join(str(p) for p in participants)
+
+    @staticmethod
+    def _deserialise_session_participants(s: str) -> set[int]:
+        """Convert a comma separated string into a set."""
+        return set(int(user_id) for user_id in s.split(",") if user_id != "")
+
+    @lock.lock_arg(NAMESPACE, "message", attrgetter("channel.id"))
+    @lock.lock_arg(NAMESPACE, "message", attrgetter("author.id"))
+    async def notify_session_participants(self, message: discord.Message) -> None:
+        """
+        Check if the message author meets the requirements to be notified.
+
+        If they meet the requirements they are notified.
+        """
+        if await _caches.claimants.get(message.channel.id) == message.author.id:
+            return  # Ignore messages sent by claimants
+
+        if not await _caches.help_dm.get(message.author.id):
+            return  # Ignore message if user is opted out of help dms
+
+        if message.content == f"{self.bot.command_prefix}close":
+            return  # Ignore messages that are closing the channel
+
+        session_participants = self._deserialise_session_participants(
+            await _caches.session_participants.get(message.channel.id) or ""
+        )
+
+        if message.author.id not in session_participants:
+            session_participants.add(message.author.id)
+
+            embed = discord.Embed(
+                title="Currently Helping",
+                description=f"You're currently helping in <#{message.channel.id}>",
+                color=discord.Colour.green(),
+                timestamp=message.created_at
+            )
+            embed.add_field(name="Conversation", value=f"[Jump to message]({message.jump_url})")
+            await message.author.send(embed=embed)
+
+            await _caches.session_participants.set(
+                message.channel.id,
+                self._serialise_session_participants(session_participants)
+            )
+
+    @commands.command(name="helpdm")
+    async def helpdm_command(
+            self,
+            ctx: commands.Context,
+            state: allowed_strings("on", "off") = None  # noqa: F821
+    ) -> None:
+        """
+        Allows user to toggle "Helping" dms.
+
+        If this is set to off the user will not receive a dm for channel that they are participating in.
+        If this is set to on the user will receive a dm for the channel they are participating in.
+        """
+        state_bool = state.lower() == "on"
+
+        requested_state_bool = state.lower() == "on"
+        if requested_state_bool == await _caches.help_dm.get(ctx.author.id, False):
+            if await _caches.help_dm.get(ctx.author.id):
```

Comment:
Might be worth saving to a variable the `await _caches.help_dm.get(ctx.author.id)` so that it isn't requested twice if the first if condition is False.